src/items/functions.md

```diff
@@ -107,35 +107,73 @@ component after the function name. This might be necessary if there is not
 sufficient context to determine the type parameters. For example,
 `mem::size_of::<u32>() == 4`.
 
-## Extern functions
+## Extern function qualifier
 
-Extern functions are part of Rust's foreign function interface, providing the
-opposite functionality to [external blocks]. Whereas external
-blocks allow Rust code to call foreign code, extern functions with bodies
-defined in Rust code _can be called by foreign code_. They are defined in the
-same way as any other Rust function, except that they have the `extern`
-qualifier.
+The `extern` function qualifier allows providing function _definitions_ that can
+be called with a particular ABI:
```

Comment:
Hmm, this seems to imply it's specific to definitions but the exact same syntax with the same default also applies to `fn` pointers.

Reply:
Not the exact same (e.g. function pointers have no identifier or body), but I see your point. Stretching this out a bit more, an also similar syntax applies to declarations in extern blocks. 

I think that the "extern block item" and the "function item" sections should just tangentially mention `extern "ABI"`, and that a different section describing the "extern qualifier" should be used to explain the whole thing. As it is done right now, we have links back and forth between both sections for no good reason.

Reply:
IMO, punt this change to another PR. This one is getting a lot of back and forth, and I think we should setting with an improved base to work on.

That said, for that section, something like this could work:

```md
The `extern` qualifier specifies the [<abbr title="Application Binary Interface>ABI</abbr>][ABI] for calling the functions they apply to. It is applicable to ...

[ABI]: https://en.wikipedia.org/ABI
```

Reply:
So I'm going to leave this as is for now . Even if it still isn't 100% clear, it is still an improvement over what we have. I've opened #655 to track my suggestion for improving this further by just explaining the `extern` qualifier for all users somewhere else, and just referencing it from the function item and extern block item sections. 

If you have any better ideas about how to explain this more clearly, we can discuss those in that issue.